net/socket/recvfrom.c

```diff
@@ -152,11 +154,39 @@ ssize_t recvfrom(int sockfd, FAR void *buf, size_t len, int flags,
 {
   FAR struct socket *psock;
   ssize_t ret;
+#ifdef CONFIG_BUILD_KERNEL
+  struct sockaddr kaddr;
```

Comment:
need change to sockaddr_storage to support IPv6 or other transport

Reply:
Thanks for pointing this out, will fix!